vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -54,16 +67,31 @@ def parse_node(node)
 
     def parse_service(service)
       new_result = parse_base_item(service)
+      container_groups = []
+
+      endpoint_container_groups = @data_index.fetch_path(
+        :container_endpoints, :by_namespace_and_name, new_result[:namespace],
+        new_result[:name], :container_groups)
+
+      unless endpoint_container_groups.nil?
+        endpoint_container_groups.each do |group|
+          cg = @data_index.fetch_path(
+            :container_groups, :by_namespace_and_name, group[:namespace],
+            group[:name])
+          container_groups << cg unless cg.nil?
+        end
```

Comment:
I used "compact" to begin with but then I thought that it needs an extra pass on the entire array to remove elements that shouldn't have been added in the first place. It seems twice less efficient to add elements that you know you don't need, and then remove them with an extra pass.

I'll add this though:

 endpoint_container_groups ||= []